### PR TITLE
wrapping btCollisionWorld and btCollisionObject in Bullet component

### PR DESCRIPTION
**Pull request by Joilnen**

### CODE REVIEW

**Comment by Joilnen** [comment]
I am getting a error in SWIG, that I can't to know to fix, I don't use SWIG, could someone to help me how to fix it ?

**Comment by paroj** [comment]
> I am getting a error in SWIG, that I can't to know to fix, I don't use SWIG, could someone to help me how to fix it ?

rename getBtWorld() in CollisionWorld to e.g. getBtCollisionWorld() to get rid of them.

However, you first should try to get a workable state for yourself.

File: Components/Bullet/src/OgreBullet.cpp [hunk 1]
```diff
@@ -693,5 +692,119 @@ void DebugDrawer::drawLine(const btVector3& from, const btVector3& to, const btV
...
+    mBtWorld->addCollisionObject(co, group, mask);
+    co->setUserPointer(new EntityCollisionListener{ent, listener});
+
+
+    // transfer ownership to node
+    auto bodyWrapper = std::make_shared<CollisionObject>(co, mBtWorld);
+    bodyWrapper->setNode(node);
+    node->getUserObjectBindings().setUserAny("BtColiisionObject", bodyWrapper);
+
+    auto ret = btTransform(convert(node->getOrientation()), convert(node->getPosition()));
```
Comment (paroj) [inline Components/Bullet/src/OgreBullet.cpp:794]: as there is no btMotionState for btCollisionObject, you will have to do this manually every time Ogre::SceneNode moves.
At this point you can just do nothing here, and fully move the updates to your code.

**Comment by paroj** [comment]
also see #2764, where I split DynamicsWorld, so you can inject your own CollisionWorld

**Comment by Joilnen** [comment]
> also see #2764, where I split DynamicsWorld, so you can inject your own CollisionWorld

Sorry but just there is a CollisionWorld, the only thing I didn't see is 
 virtual btCollisionWorld* getBtCollisionWorld() const { return mBtWorld; }, member function is that not necessary anymore ?
 can close this pull request and we go on with yours ?

**Comment by paroj** [comment]
> the only thing I didn't see is virtual btCollisionWorld* getBtCollisionWorld() const

yeah, I omitted that one as to construct CollisionWorld, you must pass an externally created btCollisionWorld. Therefore btCollisionWorld is already visible from the outside.

The getBtWorld vs getBtCollisionWorld is a SWIG issue that I did not want to reflect in the API. All this will likely be for advanced usage, so minor inconveniences are tolerable I think

**Comment by Joilnen** [comment]
> > the only thing I didn't see is virtual btCollisionWorld* getBtCollisionWorld() const
> 
> yeah, I omitted that one as to construct CollisionWorld, you must pass an externally created btCollisionWorld. Therefore btCollisionWorld is already visible from the outside.
> 
> The getBtWorld vs getBtCollisionWorld is a SWIG issue that I did not want to reflect in the API. All this will likely be for advanced usage, so minor inconveniences are tolerable I think
Ok I got it, closing this, your commit gives a good solution for that, maybe in future we  could make the component to manage and hide better bullet, making this easier to use it.

**Pull request closed without merging**